src/main/features/response/routes/your-details.ts

```diff
@@ -44,7 +47,18 @@ export default express.Router()
 
     const partyDetails: PartyDetails = plainToClass(PartyDetails, user.claim.claimData.defendant)
     if (user.responseDraft.defendantDetails.partyDetails) {
-      partyDetails.name = user.responseDraft.defendantDetails.partyDetails.name
+      switch (user.responseDraft.defendantDetails.partyDetails.type) {
```

Comment:
Looks like fix for something we missed the other day.

I wonder if we could make it even better by pointing the right type to `plainToClass(<here>, user.claim.claimData.defendant)`.

Then we could extract type from `user.claim.claimData.defendant.type`to find constructor for given party type.

To summarise something like `plainToClass(PartyDetailsFactory.constructorFor(user.claim.claimData.defendant.type), user.claim.claimData.defendant)`

Reply:
will give it a go, but do need to get this in soon for release next week +1

Reply:
Theoretically it should work. I thought that when pass any constructor for class extending from `TheirDetails` it should nicely deserialise it but I never tested subclasses with this library. Happy to leave it as is for now and rework it when we start using the library more extensively.

Reply:
subclassing does work